app/src/main/java/org/kinecosystem/kinit/viewmodel/spend/AppViewModel.kt

```diff
@@ -4,21 +4,18 @@ import android.content.Context
 import android.databinding.ObservableBoolean
 import android.databinding.ObservableInt
 import android.support.v4.view.ViewPager
-import android.util.Log
 import android.view.View
-import android.widget.Toast
 import org.kinecosystem.kinit.KinitApplication
 import org.kinecosystem.kinit.analytics.Analytics
 import org.kinecosystem.kinit.analytics.Events
-import org.kinecosystem.kinit.model.spend.EcoApplication
+import org.kinecosystem.kinit.model.spend.EcosystemApp
 import org.kinecosystem.kinit.model.spend.getHeaderImageCount
 import org.kinecosystem.kinit.model.spend.getHeaderImageUrl
 import org.kinecosystem.kinit.model.spend.isKinTransferSupported
 import org.kinecosystem.kinit.navigation.Navigator
-import org.kinecosystem.kinit.util.GeneralUtils
 import javax.inject.Inject
 
-class AppViewModel(private val navigator: Navigator, private val app: EcoApplication, context:Context) : ViewPager.OnPageChangeListener {
+class AppViewModel(private val navigator: Navigator, private val app: EcosystemApp, isAppInstalled: Boolean) : ViewPager.OnPageChangeListener {
```

Comment:
not sure it should be as param
its appview model role to do the check 
not the view that use it